BluetoothAdvertisementsKotlin/app/src/main/java/com/example/bluetoothadvertisements/MainActivity.kt

```diff
@@ -0,0 +1,108 @@
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import com.example.bluetoothadvertisements.databinding.ActivityMainBinding
+
+private const val TAG = "MainActivity"
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setSupportActionBar(findViewById(R.id.toolbar))
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+
+        if (savedInstanceState == null) {
+            verifyBluetooth()
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+
+        if (resultCode == RESULT_OK) {
+            when (requestCode) {
+                REQUEST_ENABLE_BT -> {
+                    Log.d(TAG, "onActivityResult: REQUEST_ENABLE_BT")
+                    verifyBluetooth()
+                }
+                PERMISSION_REQUEST_LOCATION -> {
+                    Log.d(TAG, "onActivityResult: PERMISSION_REQUEST_COARSE_LOCATION")
+                    verifyBluetooth()
+                }
+            }
+        }
+        super.onActivityResult(requestCode, resultCode, data)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        menuInflater.inflate(R.menu.menu_main, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        return when (item.itemId) {
+            R.id.action_settings -> true
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun setupFragments() {
+        val fragTransaction = supportFragmentManager.beginTransaction()
+        fragTransaction.replace(R.id.scanner_fragment_container, ScannerFragment())
+        fragTransaction.replace(R.id.advertiser_fragment_container, AdvertiserFragment())
+        fragTransaction.commit()
+    }
+
+    private fun verifyBluetooth() {
+        val bluetoothAdapter = (getSystemService(BLUETOOTH_SERVICE) as BluetoothManager).adapter
+
+        when {
+            bluetoothAdapter == null ->
+                showErrorText("onCreate: bluetooth not supported")
+            bluetoothAdapter.isEnabled && bluetoothAdapter.isMultipleAdvertisementSupported ->
+                setupFragments()
+            bluetoothAdapter.isEnabled && !bluetoothAdapter.isMultipleAdvertisementSupported ->
+                showErrorText("Bluetooth Advertisements are not supported.")
+            !bluetoothAdapter.isEnabled ->
```

Comment:
what is this doing exactly, requesting some permission or sending them to a settings page? Can you comment?